### PR TITLE
Static Build

### LINKED ISSUE #619: Static build
Building statically doesn't work. We should fix that.
To fix it, it'll basically consist of turning on static builds using `./b configure -i`, then fixing the errors that it makes. You MUST `./b configure --clean -- -DSTATIC_LIBRARIES=ON` to turn on static builds though, because (IIRC) once you configure without static libraries on, you can't turn them on properly (I am probably misremembering this somehow, but that command is guaranteed to set you on the right path I think).

### PR DESCRIPTION
**Pull request by CMurtagh-LGTM**
Closes #619 Trying to build libraries using static linking, rather than dynamic.

Since its not wise to statically build everything I also made it a case by case basis.

### CODE REVIEW

**Comment by CMurtagh-LGTM** [comment]
```
[419/702] Linking CXX shared library bin/lib/libinputCamera.so
FAILED: bin/lib/libinputCamera.so
: && /usr/sbin/g++ -fPIC -fPIC -mtune=generic -fPIC -mtune=generic -g   -shared -Wl,-soname,libinputCamera.so -o bin/lib/libinputCamera.so module/input/Camera/CMakeFiles/inputCamera.dir/src/Camera.cpp.o module/input/Camera/CMakeFiles/inputCamera.dir/src/time_sync.cpp.o  -Wl,-rpath,lib/:../lib/:bin/lib  /usr/local/lib/libnuclear.a  bin/lib/libnuclear_extension.so  /usr/local/lib/libaravis-0.8.a  bin/lib/libnuclear_utility.so  /usr/local/lib/libnuclear.a  /usr/local/lib/libyaml-cpp.a  /usr/local/lib/libfmt.a  /usr/local/lib/libz.a  /usr/local/lib/libaravis-0.8.a  /usr/local/lib/libgio-2.0.a  /usr/local/lib/libgmodule-2.0.a  /usr/local/lib/libgobject-2.0.a  /usr/local/lib/libglib-2.0.a  /usr/local/lib/libgthread-2.0.a  /usr/lib/libmount.so  /usr/lib/libresolv.a  /usr/local/lib/libusb-1.0.a  /usr/lib/libudev.so  /usr/local/lib/libffi.a  /usr/local/lib/libxml2.a  /usr/lib/liblzma.so  /usr/local/lib/libbacktrace.a  -ldl  bin/lib/libnuclear_message.so  shared/clock/libnubots_clock.a  /usr/local/lib/libprotobuf.a  -lpthread  /usr/local/lib/libyaml-cpp.a  /usr/local/lib/libgio-2.0.a  /usr/local/lib/libgmodule-2.0.a  /usr/local/lib/libgobject-2.0.a  /usr/local/lib/libglib-2.0.a  /usr/local/lib/libgthread-2.0.a  /usr/lib/libmount.so  /usr/lib/libresolv.a  /usr/local/lib/libusb-1.0.a  /usr/lib/libudev.so  /usr/local/lib/libffi.a  /usr/local/lib/libxml2.a  /usr/lib/liblzma.so && :
/usr/sbin/ld: /usr/lib/libresolv.a(res_send.o): warning: relocation against `in6addr_any@@GLIBC_2.2.5' in read-only section `.text'
/usr/sbin/ld: /usr/lib/libresolv.a(res_send.o): relocation R_X86_64_PC32 against symbol `in6addr_any@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/sbin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

**Comment by CMurtagh-LGTM** [comment]
TODO nubook pr

File: cmake/Modules/FindAravis.cmake [hunk 1]
```diff
@@ -1,6 +1,10 @@
 include(ToolchainLibraryFinder)
 
 find_package(glib2 REQUIRED)
+find_package(libusb1 REQUIRED)
+# find_package(zlib REQUIRED)
```
Comment (KipHamiltons) [inline cmake/Modules/FindAravis.cmake:5]: Commented

File: cmake/Modules/FindAravis.cmake [hunk 2]
```diff
@@ -10,3 +14,7 @@ ToolchainLibraryFinder(
 )
 
 target_link_libraries(Aravis::Aravis INTERFACE glib2::glib2)
+target_link_libraries(Aravis::Aravis INTERFACE libusb1::libusb1)
+# target_link_libraries(Aravis::Aravis INTERFACE zlib::zlib)
```
Comment (KipHamiltons) [inline cmake/Modules/FindAravis.cmake:18]: Here too

**Review by KipHamiltons — COMMENTED** [review]
No obvious issues I can see

**Comment by CMurtagh-LGTM** [comment]
https://github.com/NUbots/NUbook/pull/244

File: nuclear/cmake/Modules/ToolchainLibraryFinder.cmake [hunk 3]
```diff
@@ -13,12 +13,44 @@ function(ToolchainLibraryFinder)
...
+          ${PACKAGE_LINK_TYPE}
+          CACHE STRING "Choose method to link the library"
+      )
+    else()
+      set(${PACKAGE_NAME}_LINK_TYPE
+          UNKNOWN
+          CACHE STRING "Choose method to link the library"
+      )
+    endif()
+    set_property(CACHE ${PACKAGE_NAME}_LINK_TYPE PROPERTY STRINGS "SHARED" "STATIC" "MODULE" "UNKNOWN")
```
Comment (ysims) [inline nuclear/cmake/Modules/ToolchainLibraryFinder.cmake:35]: How does this work with the "UNKNOWN" type?

Comment (CMurtagh-LGTM) [reply nuclear/cmake/Modules/ToolchainLibraryFinder.cmake:35]: "UNKNOWN" will use the default, i.e. search for a `.so` and if it cannot find one search for a `.a`.

**Review by ysims — APPROVED** [review]
LGTM

**Pull request merged**